NebulaNetwork/Server.cs

```diff
@@ -66,6 +70,42 @@ public override void Start()
             PacketProcessor.SimulateLatency = true;
 #endif
 
+            if (Config.Options.EnableNgrok)
+            {
+                if (!Config.Options.NgrokAuthtoken.IsNullOrEmpty())
+                {
+                    ngrokManager = new Ngrok.NgrokManager();
+
+                    // Start this stuff in it's own thread, as we require async and we dont want to freeze up the GUI when freeze up when Downloading and installing ngrok
+                    Task.Run(async () =>
+                    {
+
+                        if (!ngrokManager.IsNgrokInstalled())
+                        {
+                            await ngrokManager.DownloadAndInstallNgrok();
+                        }
+                        var successfull = ngrokManager.StartNgrok(port, Config.Options.NgrokAuthtoken);
+                        if (!successfull)
+                        {
+                            NebulaModel.Logger.Log.Warn("Failed to start Ngrok tunnel!");
+                        }
+
+                        if (ngrokManager.IsNgrokStarted())
+                        {
+                            ngrokAddress = await ngrokManager.GetTunnelAdress();
+                        } else
+                        {
+                            NebulaModel.Logger.Log.Warn("Ngrok tunnel has exitted prematurely! Invalid authtoken perhaps?");
+                        }
+
+                    });
+                } else
+                {
+                    NebulaModel.Logger.Log.Warn("Ngrok support was enabled, however no Authtoken was provided");
+                }
+
+            }                
```

Comment:
This should probably be moved to a separate method

Reply:
Agreed, question though, should I move this to a method inside the Server.cs or just make this part of the NgrokManager (Perhaps even just make this part of the NgrokManager's constructor)?

Reply:
I think it should be in NgrokManager. I suppose it does make sense to just put it in the constructor but that part is up to you.

Reply:
I moved everything to NgrokManager